### PR TITLE
Refactor/esm

### PR DESCRIPTION
**Pull request by ANGkeith**
This is quite a big mr, but i tried to break down the changes into smaller commits  

for actual file changes, can see
https://github.com/firecow/gitlab-ci-local/pull/1331/files/d06df308016637357d191e8be74a11194fa2ad2b..8760e7c3304f39fd4604ac5f64ab65edb06f6e9b
instead, 

since the "ts2esm vooodo" commit is mostly autogenerated

### CODE REVIEW

**Comment by firecow** [comment]
@ANGkeith Can you pull master. I'll review this in this week. I really want to upgrade to ESM :smile:

**Comment by firecow** [comment]
@ANGkeith Can you pull master again, and I'll put some time into testing this thoroughly

**Review by firecow — APPROVED** [review]
@ANGkeith I've added a few commits.

Merging now :+1:

File: package.json [hunk 1]
```diff
@@ -132,6 +134,18 @@
...
+      ".ts"
+    ],
+    "transform": {
+      "^.+\\.tsx?$": [
+        "ts-jest",
+        {
+          "useESM": true
+        }
+      ],
+      "^.+\\.jsx?$": "babel-jest"
```
[the remark this answers is not in the document]
Comment (firecow) [reply package.json:147]: I just wanted the PR in master didn't want to spent too much time on it.
We might be able to get rid of it.

[the remark this answers is not in the document]
Comment (ANGkeith) [reply package.json:147]: oo ok, glad too this PR is finally in master

the rest of the changes makes sense to me, thanks for the review

**Pull request merged**